### ISSUE TITLE
Setting button to pass mouse events makes clicking difficult on Android

### ISSUE THREAD

**Issue by j-zeppenfeld** [godotengine/godot#95532]
### Tested versions

- Reproducible in: v4.3.beta3.official [82cedc83c]
- Not reproducible in: v4.2.2.stable.official [15073afe3]

### System information

Android, Samsung Galaxy S10e

### Issue description

In order to make menus touch scrollable in Android, buttons must be set to pass mouse events. In Godot 4.2, the dead zone before the button passed motion events was big enough to allow consistent pressing of the button. In Godot 4.3, this dead zone seems to be practically 0, making clicking the button all but impossible.

### Steps to reproduce

Create a simple UI with a ScrollContainer, VBoxContainer and enough Buttons to get the ScrollContainer to scroll.
Set the buttons to pass mouse events (in order to make the ScrollContainer touch scrollable).
Run the project on Android.
While scrolling works, clicking the buttons is practically impossible.

### Minimal reproduction project (MRP)

The top three buttons are set to pass mouse events and work to scroll, but are difficult to click.
The bottom three buttons are set to stop mouse events and cannot be used to scroll, but can be easily clicked.
In Godot 4.2, both scrolling and clicking are easily possible on the top three buttons.

[exittest.zip](https://github.com/user-attachments/files/16619232/exittest.zip)

**Comment by akien-mga** [comment]
May be related to https://github.com/godotengine/godot/issues/93612, cc @m4gr3d @Alex2782

**Comment by Alex2782** [comment]
> this dead zone seems to be practically 0

yes for #84138 and #84182. However, it is possible to customize `Scroll Deadzone` in the `ScrollContainer`.

![image](https://github.com/user-attachments/assets/e9719ec4-93ae-45e1-98e2-2cb616dfbdc1)

---------------------

In the next few days I will try to create PR for new default value, first I would have to check the effects on other operating systems. 

https://github.com/search?q=repo%3Agodotengine%2Fgodot+scroll_deadzone&type=code

**Comment by Alex2782** [comment]
[scroll_test.zip](https://github.com/user-attachments/files/16646439/scroll_test.zip)

@j-zeppenfeld :  I couldn't find any issues with `deadzone = 0` on my Android devices.
See video, from which `deadzone` value is everything ok on your device? (DPI / Scale / Size values?)

Perhaps we can somehow take the DPI / Scale / Size values into account for touchscreens.

<details><summary>Video: Samsung Tab S7, 12.4" display </summary>

https://github.com/user-attachments/assets/17922bf5-f198-4de6-b0a9-fc9c9fffa840
</details>

Also on my `Pixel 4a`, 5.8" display, DPI: 440, Scale: 1, Size: (2340, 1080) like in video 100% no scroll events triggered.  

------------------------------------------------------------

Note: There is a bug, already 6 years old. Further down in the `ScrollContainer` scrolling over controls does not always work, `pressed` should always work there 😃 

<details><summary>Video: issue  #22936</summary>


https://github.com/user-attachments/assets/8d811734-7ea8-442e-be5c-2d17a1a9712c



</details>

**Comment by j-zeppenfeld** [comment]
Deadzone=0 is definitely not usable on my Galaxy S10e (DPI 480 Scale 1), quick taps from an experienced user are ok at deadzone=2, for less experienced users I would probably set the deadzone to at least 4.

Here's a video of what it looks like on my device:
https://github.com/user-attachments/assets/fd081112-b0e5-45ab-bd14-ff6a80c112c8

**Comment by j-zeppenfeld** [comment]
Playing with this some more, it seems that my device is especially eager to send InputEventScreenDrag.

The absolute minimum output I can achieve for as short a tap as possible allows includes at least 5 events: an InputEventScreenDrag surrounded by press and release InputEventMouseButton and InputEventScreenTouch events.

**Comment by Alex2782** [comment]
@j-zeppenfeld If you find time please try PR #95827, with your productive projects.

or `android_debug_base_button.apk` 27 MB Download
https://drive.google.com/file/d/1vVCK1tUHzi6kloKjcY698CBDkDDrGv9q/view?usp=sharing

![Bildschirmfoto 2024-08-20 um 23 08 10](https://github.com/user-attachments/assets/98baebd7-96f9-4765-be06-c68bc5f3715c)

-----------------------

`adb logcat -s godot`

**Logs:** on my Samsung Tab S7, `pressed_tolerance = round (DPI / 50) `
```
08-20 22:52:53.344 16666 16735 I godot   : InputEventMouseMotion, drag_from:  (846.6693, 378.7633) , is_scrolling:  true
08-20 22:52:53.344 16666 16735 I godot   : 	 drag_from:  (846.6693, 378.7633)
08-20 22:52:53.344 16666 16735 I godot   : 	 get_mouse_position:  (844.67, 380.762)
08-20 22:52:53.344 16666 16735 I godot   : 	 diff:  (-1.999207, 1.998657)
08-20 22:52:53.344 16666 16735 I godot   : 	 pressed_tolerance:  7
```

**Comment by j-zeppenfeld** [comment]
> @j-zeppenfeld If you find time please try PR #95827, with your productive projects.
Yup, working great in both projects (my main pixel art project and your modified exit test project).
Thank you!

**Comment by goranovs** [comment]
Can I desperately ask somebody to give attention to this issue

**Comment by Alex2782** [comment]
@goranovs 
Is this configuration an option for you? Increase to 10.
https://github.com/godotengine/godot/issues/95532#issuecomment-2290024808

**Comment by goranovs** [comment]
Hi @Alex2782,

I've used 5 to make it usable again. Anything below 5 felt like there was a chance to miss the press, but 10 seems to be the better value—the scrolling feels more natural, and the button press works fine.

Thank you!
